addons/account/models/account_account.py

```diff
@@ -654,18 +663,22 @@ def default_get(self, default_fields):
         instead of its name. In that case, switch both fields values.
         """
         if 'name' not in default_fields and 'code' not in default_fields:
-            return super().default_get(default_fields)
-        default_name = self._context.get('default_name')
-        default_code = self._context.get('default_code')
-        if default_name and not default_code:
-            try:
-                default_code = int(default_name)
-            except ValueError:
-                pass
-            if default_code:
-                default_name = False
-        contextual_self = self.with_context(default_name=default_name, default_code=default_code)
-        return super(AccountAccount, contextual_self).default_get(default_fields)
+            res = super().default_get(default_fields)
+        else:
+            default_name = self._context.get('default_name')
+            default_code = self._context.get('default_code')
+            if default_name and not default_code:
+                with contextlib.suppress(ValueError):
+                    default_code = int(default_name)
+                if default_code:
+                    default_name = False
+            contextual_self = self.with_context(default_name=default_name, default_code=default_code)
+            res = super(AccountAccount, contextual_self).default_get(default_fields)
```

Comment:
Do we even need the early return with `ìf 'name' not in default_fields`? It's not going to change anything, except the context that won't be read... 
I'd go for simplicity in this case since it doesn't cost anything.

```suggestion
            context = {}
        else:
            default_name = self.env.context.get('default_name')
            default_code = self.env.context.get('default_code')
            if default_name and not default_code:
                with contextlib.suppress(ValueError):
                    default_code = int(default_name)
                if default_code:
                    default_name = False
            context = {'default_name': default_name, 'default_code': default_code}
        defaults = super(AccountAccount, self.with_context(**context)).default_get(default_fields)
```

Reply:
I agree this is cleaner :100: